src/rust/engine/fs/src/glob_matching.rs

```diff
@@ -37,7 +37,7 @@ lazy_static! {
   static ref MISSING_GLOB_SOURCE: GlobParsedSource = GlobParsedSource(String::from(""));
   static ref PATTERN_MATCH_OPTIONS: MatchOptions = MatchOptions {
     require_literal_separator: true,
-    ..MatchOptions::default()
+    ..MatchOptions::new()
```

Comment:
Maybe add a comment here about the issue so future readers know that the distinction between `new` and `default` matters?

Reply:
So I never really "learned" Rust. What's the normal Rustacean way to do this kinda thing. Seems weird to me that `default` and `new` don't agree, and that since the programmer wrote `new`, that'd be the "default" (no pun intended).

Reply:
Edited.

Reply:
Your perception that it is weird is a good one. The authors of that `MatchOptions` struct are actually being anti-idiomatic by having `new` and `default` disagree with each other. All the more reason for us to be explicit in this code.

Indeed, the `MatchOptions` struct only has three fields, if we are setting two of them, then we should probably just specify all three explicitly.

Reply:
+1 to explicit args

Reply:
+1. That has an added advantage of being future proof in case they change it so that new and default does agree, it won't bite us regardless of in which direction that would be fixed.

Reply:
Coolio Julio. I prefer that as well, but cargo-culted the implementation in case unpacking was trying to "future-proof" new members or something equally unknown-but-conventional 🙂